src/hotspot/share/nmt/memTracker.hpp

```diff
@@ -62,6 +61,15 @@ class MemTracker : AllStatic {
     return _tracking_level != NMT_unknown;
   }
 
+  // We need this to avoid attempting to lock NmtVirtualMemory_lock during VM init (before mutexes are ready or current thread has been assigned).
+  static inline bool is_done_bootstrap()  {
+    return _done_bootstrap;
+  }
+
+  static inline void set_done_bootstrap()  {
+    _done_bootstrap = true;
+  }
+
```

Comment:
I think I would prefer "bootstrap_done" rather than "done_bootstrap" throughout.  But you should get
opinions from some of the runtime folks like @coleenp and @dholmes-ora .

Reply:
Yes, I agree, bootstrap_done reads better.

Reply:
There's a few other is_bootstrapping() predicates in the code like Universe::is_bootstrapping.  This might read nicely as is_bootstrapping_done() and the instance is _bootstrapping_done.

Reply:
Ok I've updated everything to use "bootstrapping_done" instead of "done_bootstrap"

Reply:
Arguably, MemTracker is the wrong place anyway since the problem it tries to solve is not tied to NMT initialization.

The thought behind the "bootstrap_done" was to make visible that it is safe to not lock since we still run effectively single-threaded. Hotspot loading and CreateJavaVM may happen on different threads, but not at the same time.

We can delay real locking until other threads are starting up. That is somewhere in the middle of VM initialization, but long after NMT has booted. But we can do it also as soon as Mutex initialization is done. 

So we could either move this to the mutex layer and give us a "MutexLocker::all_initialised". In that case, the implicit assumption is that as long as the JVM initializes this before other threads are started. Or, we could move it to the thread subsystem and give us a marker for when we are not single-threaded anymore. I like the former better.

Reply:
> Hotspot loading and CreateJavaVM may happen on different threads

??? What are you referring to as "hotspot loading"? To me that _is_ CreateJavaVM.

> Or, we could move it to the thread subsystem

We already have `Threads::number_of_threads()1.